lib/AST/ASTImporter.cpp

```diff
@@ -4549,18 +4569,24 @@ Decl *ASTNodeImporter::VisitClassTemplateDecl(ClassTemplateDecl *D) {
 
   ToTemplated->setDescribedClassTemplate(D2);
 
-  if (ToTemplated->getPreviousDecl()) {
-    assert(
-        ToTemplated->getPreviousDecl()->getDescribedClassTemplate() &&
-        "Missing described template");
-    D2->setPreviousDecl(
-        ToTemplated->getPreviousDecl()->getDescribedClassTemplate());
-  }
   D2->setAccess(D->getAccess());
   D2->setLexicalDeclContext(LexicalDC);
-  if (!IsFriend)
+
+  if (D->getDeclContext()->containsDeclAndLoad(D))
+    DC->addDeclInternal(D2);
+  if (DC != LexicalDC && D->getLexicalDeclContext()->containsDeclAndLoad(D))
     LexicalDC->addDeclInternal(D2);
 
+  if (FoundTemplate) {
+    auto *Recent =
+        const_cast<ClassTemplateDecl *>(FoundTemplate->getMostRecentDecl());
+    D2->setPreviousDecl(Recent);
```

Comment:
I think the decl chains of template and templated Decls should match. Probably add assert here?
`Recent->getTemplatedDecl() == ToTemplated->getPreviousDecl()`